api/client.go

```diff
@@ -457,3 +469,35 @@ func HandleConnError(err error) error {
 
 	return nil
 }
+
+func serverIncompatible(ctx context.Context, client Client) error {
+	srv, err := client.GetServerVersion(ctx)
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("check server version")
+		return nil // result inconclusive
+	}
+	ahead, err := versionAhead(client.Version, srv.Version)
+	if err != nil {
+		logging.L.Debug().Err(err).Msg("failed to compare client version to server")
+		return nil // result inconclusive
+	}
+	if ahead {
+		return fmt.Errorf("client version (%s) is ahead of server version (%s), please upgrade your server", apiVersion, srv.Version)
+	}
+	return nil
+}
+
+// versionAhead returns if the current semantic version is ahead of the comparison version
+func versionAhead(current, compare string) (bool, error) {
+	c, err := semver.NewConstraint(fmt.Sprintf("> %s", compare))
+	if err != nil {
+		return false, fmt.Errorf("failed to check api semantic version: %w", err)
+	}
+
+	v, err := semver.NewVersion(current)
+	if err != nil {
+		return false, fmt.Errorf("could not create semantic server version: %w", err)
+	}
+
+	return c.Check(v), nil
```

Comment:
Could we use https://pkg.go.dev/github.com/Masterminds/semver/v3#Version.GreaterThan instead of `NewConstraint` ?